sort/mergesort.go

```diff
@@ -60,3 +60,29 @@ func MergeIter[T constraints.Ordered](items []T) []T {
 	}
 	return items
 }
+
+// MergeParallel Perform merge sort on a slice using goroutines
+func MergeParallel[T constraints.Ordered](items []T) []T {
+
```

Comment:
Non-blocking but this newline here is really not looking nice to me 😂

Reply:
This should be ok now